src/value-types.js

```diff
@@ -0,0 +1,218 @@
+var execute = require('./execute');
+
+var Value = function Value(value) {
+    this.value = value;
+};
+
+Value.prototype.equals = function(b) {
```

Comment:
I like to pretend javascript is an entirely functional language and just make everything an object literal with a load of functions.